packages/wrangler/src/pages/publish.tsx

```diff
@@ -249,13 +248,44 @@ export const Handler = async ({
 		}
 	}
 
+	let _headers: string | undefined,
+		_redirects: string | undefined,
+		_routesGenerated: string | undefined,
+		_routesCustom: string | undefined,
+		_workerJS: string | undefined;
+
+	try {
```

Comment:
In theory, what I would have wanted here is to attempt building Functions first, before we read all these files below. If `buildFunctions` fails we want to `exit`, in which case, we've done all this work for nothing. However, to `buildFunctions` we must first attempt to read `_worker.js`, because `buildFunctions` is called only if the `/functions` directory exists and no `_worker.js` file was provided. That would have scattered these
```
try {
readFileSync()
} catch{}
```
blocks all over the place, which is not optimal for readability IMHO.

I don't know if `readFileSync` is an expensive operation. If it is, or if we think that this can potentially cause perf issues, then I will adjust the code accordingly. If not, then I'd rather keep things as is, for readability reasons. IMHO having all file reads grouped together rather than scattered in various places, makes things more tidy and easy to follow 🙇

Reply:
`readFileSync` on these small files is negligible. Totally fine how you've done it :)